zebra-state/src/service.rs

```diff
@@ -270,6 +270,95 @@ impl StateService {
             state: IterState::NonFinalized(hash),
         }
     }
+
+    /// Return an iterator over the finalized blocks of the chain
+    /// starting (and descending) from the block identified by `height`.
+    ///
+    /// The block identified by `height` is included in the chain of blocks yielded
+    /// by the iterator.
+    pub fn chain_finalized(&self, height: block::Height) -> Iter<'_> {
+        Iter {
+            service: self,
+            state: IterState::Finalized(height),
+        }
+    }
+
+    /// Return a list of blocks ahead given a list of blocks.
+    ///
+    /// 1. Make sure we have the list of blocks provided in our chain.
+    /// 2. Get a max of 500 hashes ahead of the provided tip.
```

Comment:
Let's document the specific protocol requirements here:
```suggestion
    /// Return a list of block hashes in the best chain, following the first matching hash in `known_blocks`.
    ///
    /// Starts from the first matching hash in the best chain, ignoring all other hashes in `known_blocks`.
    /// Only matches and returns hashes from the best chain, including non-finalized blocks.
    /// Stops the list of hashes after:
    ///   * adding the non-finalized best tip,
    ///   * adding the stop hash to the list, if it is in the best chain, or
    ///   * adding 500 hashes to the list.
```